Add basic fast search engine

aeirya opened the pull request:
available algorithms:
- TF-IDF
- SBERT

how to use:
```py
from search_engine import SearchEngine
from database import get_selected_nodes as id2doc

ids = search.query('a sample query text')
print(id2doc(ids))

ids = search.related_docs(doc_id= 1)
print(id2doc(ids))
```

aeirya committed:
Add basic fast search engine
available algorithms:
- TF-IDF
- SBERT

itsalirezam committed:
Update Pipfile

itsalirezam committed:
Merge branch 'develop' into fast-search

itsalirezam commented:
<img width="1085" alt="image" src="https://github.com/wggb/wiki-website/assets/140726477/6a8abd22-93b7-4ca5-a349-0f82ed574266">

aeirya committed:
Fix bug in loading database

aeirya commented:
> <img alt="image" width="1085" src="https://private-user-images.githubusercontent.com/140726477/293345972-6a8abd22-93b7-4ca5-a349-0f82ed574266.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDM4NDU2NjIsIm5iZiI6MTcwMzg0NTM2MiwicGF0aCI6Ii8xNDA3MjY0NzcvMjkzMzQ1OTcyLTZhOGFiZDIyLTkzYjctNGNhNS1hMzQ5LTBmODJlZDU3NDI2Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjMxMjI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyOVQxMDIyNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMjg0YmFjZWE5YmIwNzkxY2E2YTQ0YTQxZDJlMmRjMzhkMzljNTkzNjgyYWQ1YzAyOTA4MzljZTliMDk0Y2VmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.EX1_xVtsK4_27mLxyJeqSSCPA99CpQlfAxcTjC3j9-U">

Fixed.

aeirya committed:
Fixed possible bug
- TFIDF errors in small number of docs

itsalirezam committed:
update

itsalirezam committed:
Fix results bug

aeirya committed:
Fix bug for low number of documents

aeirya committed:
Final bug fix!

aeirya committed:
Add feature: Search engine live indexing
- Engine automatically restarts indexing if there are enough added new documents

itsalirezam committed:
Update app.py

aeirya committed:
Fix bug: Uninitialized search engine

aeirya committed:
Merge branch 'fast-search' of https://github.com/wggb/wiki-website into fast-search

sadra1f committed:
Update Pipfile.lock

sadra1f committed:
Cleanup user interface

sadra1f committed:
Change website name

sadra1f committed:
Update transitions

sadra1f committed:
Restructure the project

sadra1f committed:
Move js and css files to assets directory

sadra1f committed:
Update related topics
